examples/recurrent.py

```diff
@@ -0,0 +1,170 @@
+'''
+Recurrent network example.  Trains a bidirectional vanilla RNN to output the
+sum of two numbers in a sequence of random numbers sampled uniformly from
+[0, 1] based on a separate marker sequence.
+'''
+
+from __future__ import print_function
+
+
+import numpy as np
+import theano
+import theano.tensor as T
+import lasagne
+
+
+# Min/max sequence length
+MIN_LENGTH = 50
+MAX_LENGTH = 55
+# Number of units in the hidden (recurrent) layer
+N_HIDDEN = 100
+# Number of training sequences in each batch
+N_BATCH = 100
+# Optimization learning rate
+LEARNING_RATE = .001
+# All gradients above this will be clipped
+GRAD_CLIP = 100
+# How often should we check the output?
+EPOCH_SIZE = 100
+# Number of epochs to train the net
+NUM_EPOCHS = 10
+
+
+def gen_data(min_length=MIN_LENGTH, max_length=MAX_LENGTH, n_batch=N_BATCH):
+    '''
+    Generate a batch of sequences for the "add" task, e.g. the target for the
+    following
+
+    ``| 0.5 | 0.7 | 0.3 | 0.1 | 0.2 | ... | 0.5 | 0.9 | ... | 0.8 | 0.2 |
+      |  0  |  0  |  1  |  0  |  0  |     |  0  |  1  |     |  0  |  0  |``
+
+    would be 0.3 + .9 = 1.2.  This task was proposed in [1]_ and explored in
+    e.g. [2]_.
+
+    Parameters
+    ----------
+    min_length : int
+        Minimum sequence length.
+    max_length : int
+        Maximum sequence length.
+    n_batch : int
+        Number of samples in the batch.
+
+    Returns
+    -------
+    X : np.ndarray
+        Input to the network, of shape (n_batch, max_length, 2), where the last
+        dimension corresponds to the two sequences shown above.
+    y : np.ndarray
+        Correct output for each sample, shape (n_batch,).
+    mask : np.ndarray
+        A binary matrix of shape (n_batch, max_length) where ``mask[i, j] = 1``
+        when ``j <= (length of sequence i)`` and ``mask[i, j] = 0`` when ``j >
+        (length of sequence i)``.
+
+    References
+    ----------
+    .. [1] Hochreiter, Sepp, and Jurgen Schmidhuber. "Long short-term memory."
```

Comment:
It should be "Jürgen", not "Jurgen". You will need to begin the file with a proper encoding definition then, like this:
```
#!/usr/bin/env python
# -*- coding: utf-8 -*-
```
While we're at it, it would also be nice to set the executable bit for the script (`chmod +x recurrent.py`), that's retained by `git`.